sklearn/tests/test_common.py

```diff
@@ -82,6 +80,25 @@ def test_all_estimators():
         yield check_parameters_default_constructible, name, Estimator
 
 
+def test_non_meta_estimators():
+    # input validation etc for non-meta estimators
+    # FIXME these should be done also for non-mixin estimators!
+    estimators = all_estimators(type_filter=['classifier', 'regressor',
+                                             'transformer', 'cluster'])
+    for name, Estimator in estimators:
+        if name not in CROSS_DECOMPOSITION + ['Imputer']:
+            # Test that all estimators check their input for NaN's and infs
+            yield check_estimators_nan_inf, name, Estimator
+
+        if (name not in ['CCA', '_CCA', 'PLSCanonical', 'PLSRegression',
+                         'PLSSVD', 'GaussianProcess']):
+            # FIXME!
+            # in particular GaussianProcess!
+            yield check_estimators_overwrite_params, name, Estimator
+        if hasattr(Estimator, 'sparsify'):
```

Comment:
I guess I did it here to be equivalent to the previous test. I guess adding it to the regressor  and classifier tests would be possible. But maybe it will be added to other estimators in the future? Does it hurt here?